workflow/controller/operator.go

```diff
@@ -3931,7 +3931,7 @@ func (woc *wfOperationCtx) setExecWorkflow(ctx context.Context) error {
 		// Validate the execution wfSpec
 		err := waitutil.Backoff(retry.DefaultRetry,
 			func() (bool, error) {
-				validationErr := validate.ValidateWorkflow(wftmplGetter, cwftmplGetter, woc.wf, validateOpts)
+				validationErr := validate.ValidateWorkflow(wftmplGetter, cwftmplGetter, woc.wf, woc.controller.Config.WorkflowDefaults, validateOpts)
```

Comment:
In the Controller, this is normally handled via `setWorkflowDefaults` above and the difference between `woc.wf` and `woc.execWf`.

That's probably why this worked in direct submissions but not cron ones or server ones, if I had to guess. Also the cron part means this doesn't just affect the server, if I'm not mistaken

Reply:
I'm not sure what you mean by "direct submissions", but if you mean submitting workflows via a `POST` request to `/api/v1/workflows`, then that too is broken. The E2E tests I updated in this PR checks for that. I created a branch of `main` that cherry-picks just the tests so you can see for yourself: https://github.com/argoproj/argo-workflows/compare/main...MasonM:argo-workflows:test-main-10946

Run `make TestSubmitWorkflowTemplateWorkflowMetadataSubstitution` and you should see the test fail with the following error:
```
workflow-template-submittable-v5xfw : Failed invalid spec: templates.whalesay-template: failed to resolve {{workflow.labels.default-label}}
Did not find node
```

Reply:
> I'm not sure what you mean by "direct submissions"

Sorry I meant through the k8s API / `kubectl` directly. Versus "server ones" I meant as through the API that are ofc failing. "cron ones" I meant bc they do go through the same `Submit` method as this PR shows, meaning it would be affecting the Controller too

Reply:
Oh, okay. Yes, it looks like Cron workflows are affected too, so I updated the title to remove the `(server)` scope.